### PR TITLE
move trust for ProductBinaries to Unrestricted.

### LINKED ISSUE #5485: unsigned modules should not load with `AllSigned` policy
Steps to reproduce
------------------

```powershell
Set-ExecutionPolicy -Policy AllSigned -scope process
import-module pester -force  #Import a module which lives in $PSHome
```

But if you copy Pester out of $PSHome it will not load

Expected behavior
-----------------

image

Actual behavior
---------------
module loads without error

Notes
-------

PS 5.1 has the same behavior

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.0.0-rc
PSEdition                      Core
GitCommitId                    v6.0.0-rc
OS                             Microsoft Windows 6.1.7601 S
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by TravisEz13**
fixes #5485 
* Causes Restricted and AllSigned execution policies to no longer trust PSHome by default.
* To prevent behavior changes to Unrestricted, the trust for PShome was added here.
* Update the description of Unrestricted to indicate that it requires the shell (an existing requirement.)

### CODE REVIEW

File: src/System.Management.Automation/security/SecurityManager.cs [hunk 1]
```diff
@@ -189,6 +185,10 @@ private bool CheckPolicy(ExternalScriptInfo script, PSHost host, out Exception r
 #endif
             if (_executionPolicy == ExecutionPolicy.Unrestricted)
             {
+                // Product binaries are always trusted
+                if (SecuritySupport.IsProductBinary(path))
```
Comment (daxian-dbw) [inline src/System.Management.Automation/security/SecurityManager.cs]: If the execution policy is unrestricted, then I think we don't need this check anymore.

Comment (daxian-dbw) [reply src/System.Management.Automation/security/SecurityManager.cs]: Chatted with Travis offline. This check is needed to bypass files downloaded from internet to `$PSHome\Module`.
Need to add a comment about this.

**Review by daxian-dbw — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
Can we add a test?

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 [hunk 2]
```diff
@@ -910,6 +938,46 @@ ZoneId=$FileType
...
+                    error = $error
+                }
+                @{
+                    testScript = $LocalSignedModule
+                    expected = $null
+                    error = $null
+                }
+            )
+
+            It "$TestTypePrefix Running <testScript> Module should return <error>" -TestCases $testDate  {
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: typo: should be `$testData`

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 [hunk 3]
```diff
@@ -910,6 +938,46 @@ ZoneId=$FileType
...
+
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: Missing `param` block

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 [hunk 4]
```diff
@@ -910,6 +938,46 @@ ZoneId=$FileType
...
+                $scriptName = $testScript
+
+                $exception = $null
+                try
+                {
+                    & $scriptName
+                }
+                catch
+                {
+                    $exception = $_
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: change to use `ShouldBeErrorId`?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 (hunk 4 above)
Comment (TravisEz13) [reply test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: Sorry, I accidentally pushed an incomplete commit.  I have updated the PR.

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 [hunk 5]
```diff
@@ -813,102 +904,134 @@ ZoneId=$FileType
...
+                    testScript = $MyComputerSignatureCorruptedScript
+                    error = $error
+                }
+                @{
+                    testScript = $MyComputerUnsignedScript
                     error = $error
                 }
+
             )
-            foreach($testScript in $testScripts) {
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: Note, $testScripts was not defined in this scope, so nothing was being run.

**Comment by TravisEz13** [comment]
Note, issue #5559 was filed when adding tests due to an issue with the tests.  I have marked the affected tests to be skipped.

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 [hunk 6]
```diff
@@ -813,102 +904,134 @@ ZoneId=$FileType
...
+                @{
+                    testScript = $IntranetSignedScript
+                    error = $null
+                }
+                @{
+                    testScript = $InternetSignedScript
+                    error = $null
+                }
+            )
+            It "$TestTypePrefix Running <testScript> Script should throw '<error>'" -TestCases $skipTestData -Skip  {}
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: Should this be `Pending` and not `Skipped`?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**